asdf/_asdf.py

```diff
@@ -75,8 +75,8 @@ def __init__(
         ignore_version_mismatch=True,
         ignore_unrecognized_tag=False,
         ignore_implicit_conversion=NotSet,
-        copy_arrays=False,
         memmap=NotSet,
+        copy_arrays=NotSet,
```

Comment:
Since we're not using a `*` here (to make these keyword only arguments) the order of the kwargs matters on the (very) off chance someone is calling this will all positioned arguments. Would you move `copy_arrays` back before `memmap`?